overlord/hookstate/ctlcmd/helpers.go

```diff
@@ -103,7 +103,14 @@ func queueCommand(context *hookstate.Context, tts []*state.TaskSet) error {
 	}
 
 	for _, ts := range tts {
-		ts.WaitAll(state.NewTaskSet(tasks...))
+		for _, t := range tasks {
+			// queue service command after all tasks, except for mark-seeded which must come after service commands
+			if t.Kind() == "mark-seeded" {
```

Comment:
We'll need to add "set-model" here as well, I'm slightly worried about hard coding this here.

This is much more complicated but maybe we should:

* add Final() bool to Task
* add SetFinal(bool)
* error in WaitFor if the task to wait for is Final()
* check for Final here

SetFinal would be defined as "SetFinal sets whether a task is final in a change which means not further change work should be performed afterward, usually these are tasks that commit a full system transition."

Then when we add "mark-seeded" or "set-model" we would do SetFinal(true) on them.

Reply:
Yes, I omitted "set-mode" for now with intent to do it as a followup if we're happy with the approach.

What you suggest is definitely a much cleaner approach, I also understand the concern, on the other hand I don't think what we do here by special-casing some critical tasks is unique, we do similiar checks in many places, in particular around conflict checks.
With erroring in `WaitFor` do you mean changing WaitFor (and as a consequence WaitAll) to return `error`? I think that would be the "complicated" part of the change, in the sense that it would touch many places; the rest is rather straightforward.

Reply:
Yes but still in general we are trying to depend on specific task kinds (except where we create them), in just a few places as possible.

Let's try a middle ground for now,  let's add to snapstate/conflict.go

```
// FinalTasks are task kinds for final tasks in a change which means no further change work should be performed afterward, usually these are tasks that commit a full system transition.
var FinalTasks = []string{"mark-seeded", "set-model"}
```

snapstate is slightly the wrong place for that, but we can always move things to `overlord/conflict` if the we feel too bothered.

Reply:
Done.